Endpoint parity DELETE /groups/{group_id_list}

mjholder opened the pull request:
# Overview
This PR is being created to address [RHINENG-15839](https://issues.redhat.com/browse/RHINENG-15839).
(A description of your PR's changes, along with why/context to the PR, goes here.)

I didn't add any unit tests since we aren't modifying data until we have the listener implemented.

## PR Checklist

- [ ] Keep PR title short, ideally under 72 characters
- [ ] Descriptive comments provided in complex code blocks
- [ ] Include raw query examples in the PR description, if adding/modifying SQL query
- [ ] Tests: validate optimal/expected output
- [ ] Tests: validate exceptions and failure scenarios
- [ ] Tests: edge cases
- [ ] Recovers or fails gracefully during potential resource outages (e.g. DB, Kafka)
- [ ] Uses [type hinting](https://docs.python.org/3/library/typing.html), if convenient
- [ ] Documentation, if this PR changes the way other services interact with host inventory
- [ ] Links to related PRs

## Secure Coding Practices Documentation Reference

You can find documentation on this checklist [here](https://github.com/RedHatInsights/secure-coding-checklist).

## Secure Coding Checklist

- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [ ] General Coding Practices

thearifismail reviewed (commented):
The PR LGTM.  Shouldn't there be test?

kruai reviewed (changes requested):
Marking this as "request changes" until a test is added

mjholder commented:
I added some tests. I also added a NO_CONTENT response with the flag on. I know we are planning on waiting for an RBAC response in some fashion (listener or something else), but I didn't want to have the function return _nothing_ with the flag on.

kruai reviewed (approved):
Nice, thanks!

thearifismail reviewed (commented):
LGTM!

kruai commented:
Turns out there is an issue with with operating_system tests in the required PR check, so it should be fine to merge this.
